### PR TITLE
Add scheduler ledger with worker execution and tests

### PR DESCRIPTION
**Pull request by owlsand**
## Summary
- add scheduler ledger migration, SQLAlchemy models, cron utilities, and worker runner for deterministic task execution
- implement handler contract, example handler, and operational scripts for registration, shadow backfills, and cutover
- cover schema, idempotency, retry, concurrency, and DST edge cases with pytest fixtures and unit tests

## Testing
- `pytest -q`

------
https://chatgpt.com/codex/tasks/task_e_68c8dfe77bd4832ba9aa9422cbefc0dc

### CODE REVIEW

File: app/worker/runner.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    now: Optional[datetime] = None,
+) -> int:
+    """Claim and process a batch of occurrences."""
+
+    now = now or _now()
+    with session_scope(session_factory) as session:
+        claimed = _claim_occurrences(session, worker_id, WORKER_CLAIM_BATCH, now)
+    processed = 0
+    for claim in claimed:
+        processed += _process_occurrence(claim.id, worker_id, resolver, session_factory)
```
[the remark this answers is not in the document]
Comment (owlsand) [reply app/worker/runner.py:247]: @codex fix comments

**Comment by owlsand** [comment]
@codex review

File: app/worker/runner.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+        handler = resolver(task.handler)
+        ctx = {
+            "occurrence_id": occurrence.id,
+            "task_id": task.id,
+            "name": task.name,
+            "scheduled_for": occurrence.scheduled_for,
+            "now": started_at,
+        }
+        payload = dict(task.payload or {})
```
[the remark this answers is not in the document]
Comment (owlsand) [reply app/worker/runner.py:191]: @codex fix comments

**Comment by owlsand** [comment]
@codex review

File: migrations/001_scheduler_ledger.sql [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  attempt_no INTEGER NOT NULL CHECK (attempt_no >= 1),
+  trigger_type TEXT NOT NULL DEFAULT 'schedule' CHECK (trigger_type IN ('schedule','retry','manual')),
+  status TEXT NOT NULL CHECK (status IN ('queued','running','succeeded','failed','canceled','timed_out')),
+  worker_id TEXT,
+  started_at TIMESTAMPTZ,
+  finished_at TIMESTAMPTZ,
+  result JSONB,
+  error JSONB,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  UNIQUE (occurrence_id, attempt_no)
```
[the remark this answers is not in the document]
Comment (owlsand) [reply migrations/001_scheduler_ledger.sql:66]: @codex fix comments
